src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+    let currentSeparator;
+
+    if (separator === undefined) {
+      currentSeparator = ',';
+    } else if (separator === null) {
+      currentSeparator = 'null';
+    } else {
+      currentSeparator = separator.toString();
+    }
+
+    for (const element of this) {
+      if (element === undefined || element === null) {
+        str += currentSeparator;
+      } else {
+        str += element + currentSeparator;
+      }
```

Comment:
you can get rid of the extra `else` construct here, if you change the first condition a bit:

 ```
for (const element of this) {
      if (element !== undefined && element !== null) {
        str += element;
      }

      str += currentSeparator;
    }
```